tests/common/beamlines/test_beamline_utils.py

```diff
@@ -134,3 +136,69 @@ def test_wait_for_v2_device_connection_passes_through_timeout(kwargs, expected_t
         mock=ANY,
         timeout=expected_timeout,
     )
+
+
+def dummy_mirror() -> FocusingMirror:
+    mirror = MagicMock(spec=FocusingMirror)
+    connect = AsyncMock()
+    mirror.connect = connect
+
+    def set_name(name: str):
+        mirror.name = name  # type: ignore
+
+    mirror.set_name.side_effect = set_name
+    mirror.set_name("")
+    return mirror
+
+
+def test_device_controller_name_propagated():
+    @beamline_utils.device_factory()
+    def device() -> FocusingMirror:
+        return dummy_mirror()
+
+    mirror = device(name="foo")
+    assert mirror.name == "foo"
+
+
+def test_device_controller_connection_is_lazy():
+    @beamline_utils.device_factory()
+    def device() -> FocusingMirror:
+        return dummy_mirror()
+
+    mirror = device(name="foo")
+    assert mirror.connect.call_count == 0  # type: ignore
+
+
+def test_device_controller_eager_connect(RE):
+    @beamline_utils.device_factory(mock=True)
+    def device() -> FocusingMirror:
+        return dummy_mirror()
+
+    mirror = device(connect_immediately=True)
+    assert mirror.connect.call_count == 1  # type: ignore
+
+
+def test_multiple_layers_of_lru_caching_does_not_affect_device():
```

Comment:
Should: I think it would be good to test that the returned devices are the same without having the extra cache here too

Reply:
That's implicitly covered by `test_device_factory_can_rename` but I think you may be right that it deserves its own test